### PR TITLE
[18.09] Fix bam upload resulting in 'auto' datatype

### LINKED ISSUE #6769: 18.09 pre-release on Main: Uploaded BAMs using autodetect given datatype "auto"
test history: https://usegalaxy.org:/u/jen/h/test-history-bam-autodected-upload

data1 == this problem

data2 == copy of data1 after edit attributes > datatype > change to bam

gtf, sam, fastqsanger all autodetect correctly

ping @guerler

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes #6769

### CODE REVIEW

**Comment by jmchilton** [comment]
Crap - thanks for the fix.

**Comment by martenson** [comment]
lol, nice one

File: test/api/test_tools_upload.py [hunk 1]
```diff
@@ -240,6 +240,22 @@ def test_upload_dbkey(self):
             datasets = run_response.json()["outputs"]
             assert datasets[0].get("genome_build") == "hg19", datasets[0]
 
+    @uses_test_history(require_new=False)
+    def test_fetch_bam_file(self, history_id):
+        bam_path = TestDataResolver().get_filename("1.bam")
+        details = self._upload_and_get_details(open(bam_path, "rb"),
+                                               api="fetch",
+                                               history_id=history_id,
+                                               assert_ok=False)
```
Comment (nsoranzo) [inline test/api/test_tools_upload.py]:
```python
        with open(bam_path, "rb") as fh:
            details = self._upload_and_get_details(fh, api="fetch", history_id=history_id, assert_ok=False)
```

File: test/api/test_tools_upload.py [hunk 2]
```diff
@@ -240,6 +240,22 @@ def test_upload_dbkey(self):
...
+        assert details["state"] == "ok"
+        assert details["file_ext"] == "bam", details
+
+    def test_upload_bam_file(self):
+        bam_path = TestDataResolver().get_filename("1.bam")
+        details = self._upload_and_get_details(open(bam_path, "rb"), file_type="auto")
```
Comment (nsoranzo) [inline test/api/test_tools_upload.py]:
```python
        with open(bam_path, "rb") as fh:
            details = self._upload_and_get_details(open(bam_path, "rb"), file_type="auto")
```

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**